src/System.Xml.XmlSerializer/src/System/Xml/Serialization/XmlSerializationReader.cs

```diff
@@ -1062,11 +1142,25 @@ protected void ReadEndElement()
 
         private object ReadXmlNodes(bool elementCanBeType)
         {
+            ArrayList xmlNodeList = new ArrayList();
```

Comment:
This is just existing code being put back, rather than new code being written, right?  If it's new, we should use the generic collections rather than the non-generic ones.  But if it's just code being ported, I understand the desire to keep it as it was, at least for now.